pgjdbc/src/main/java/com/yugabyte/ysql/ClusterAwareLoadBalancer.java

```diff
@@ -144,14 +161,17 @@ public boolean needsRefresh() {
           PSQLState.UNKNOWN_STATE, e);
     }
 
+    currentPublicIps.clear();
```

Comment:
Why currentPublicIps need to be class member variable? Because I was expecting then even currentPrivateIps should be the same way. But I see they are not. Any reason? I think looking at the code they can be passed around in function arguments.

Reply:
Noting it here for the record:
`currentPublicIps` is made member variable so that it can be accessed in another method. But `currentPrivateIps` is not needed to be accessed outside this method. It's cleared in this method itself (before reading the results).

The logic of iterating over the list of servers until a successful connection is established happens at a layer above this method. Did not want to touch that code at this point.